components/brave_wallet/browser/fil_requests.cc

```diff
@@ -101,7 +114,12 @@ absl::optional<std::string> getSendTransaction(const std::string& signed_tx) {
   dict.Set("params", std::move(params));
   dict.Set("id", 1);
 
-  return GetJSON(dict);
+  auto json = GetJSON(dict);
+  // Use substring replace instead of deserializing signed_tx to prevent
+  // long to double convertion.
+  base::ReplaceFirstSubstringAfterOffset(&json, 0, "\"{signed_tx}\"",
```

Comment:
This is not related to the rust signing code. Problem is that signed transaction which is passed as a string everywhere has fields that have uint64 type.
So we inplace this string without converting it to base::Value since this could break some fields(convert them to double).
Similar approach is used in FilTransaction::GetSignedTransaction.
Alternatively we can switch to using base::Value with string uint64 fields when passing signedTx verywhere, but this will require notable refactoring.

Reply:
I think you're misunderstanding what I'm saying. I don't think this is a good way to convert fields to strings and afaik we already have a way to do this conversion by running it through the rust parser and telling it the field(s) we want to convert

Reply:
Do you mean something like json::convert_string_value_to_uint64 but to dict instead? There is no such method ATM.